File: p3semanaz2-java/Articulo.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.p3semanaz1;

public class Articulo extends Publicacion {
    
    public Articulo(String nombre, String autor, int anho,
            String revista, int volumen, int numero, String mes){
        super(nombre,autor,anho);
        this.revista=revista;
        this.volumen=volumen;
        this.numero=numero;
        this.mes=mes;
    }
    
    private String revista;
    private int volumen;
    private int numero;
    private String mes;
    
    public String getRevista() {
        return revista;
    }
    
    public void setRevista(String revista) {
        this.revista = revista;
    }
    
    public int getVolumen() {
        return volumen;
    }
    
    public void setVolumen(int volumen) {
        this.volumen = volumen;
    }
    
    public int getNumero() {
        return numero;
    }
    
    public void setNumero(int numero) {
        this.numero = numero;
    }
    
    public String getMes() {
        return mes;
    }
    
    public void setMes(String mes) {
        this.mes = mes;
    }
    
    @Override
    public String toString(){
        String referencia = " " + mes;
        referencia += ". " + revista ;
        referencia += ", vol. " + volumen;
        referencia += ", no. " + numero;
        referencia += ".";
        return super.toString()+referencia;
    }
    
    @Override
    public void metodoAbstracto(){
        
    }
}

File: p3semanaz2-cs/p3semanaz1/Articulo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace p3semanaz1
{
    public class Articulo: Publicacion
    {
        public Articulo(string nombre, string autor, int anho, 
            string revista, int volumen, int numero, string mes) : base(nombre,
                autor,anho)
        {
            this.revista = revista;
            this.volumen = volumen;
            this.numero = numero;
            this.mes = mes;
        }

        private string revista;
        private int volumen;
        private int numero;
        private string mes;

        public string Revista { get => revista; set => revista = value; }
        public int Volumen { get => volumen; set => volumen = value; }
        public int Numero { get => numero; set => numero = value; }
        public string Mes { get => mes; set => mes = value; }

        public override string ToString()
        {
            string referencia = " " + mes;
            referencia += ". " + revista ;
            referencia += ", vol. " + volumen;
            referencia += ", no. " + numero;
            referencia += ".";
            return base.ToString()+referencia;
        }

        public override void MetodoAbstracto()
        {
            return;
        }
    }
}
